common/lib/dependabot/errors.rb

```diff
@@ -5,6 +5,210 @@
 require "dependabot/utils"
 
 module Dependabot
+  # rubocop:disable Metrics/MethodLength
+  def self.fetcher_error_details(error)
+    case error
+    when Dependabot::ToolVersionNotSupported
+      {
+        "error-type": "tool_version_not_supported",
+        "error-detail": {
+          "tool-name": error.tool_name,
+          "detected-version": error.detected_version,
+          "supported-versions": error.supported_versions
+        }
+      }
+    when Dependabot::BranchNotFound
```

Comment:
I think that it would be ideal if each of these exceptions produced these via their constructors and exposed a dedicated method to fetch them. If we had that, we would not need this mapping at all

Reply:
That's neat, I think it'd be a cool follow up to this PR!